### ISSUE TITLE
ugly bug with documented plugins

### ISSUE THREAD

**Issue by Bader-Idris** [ionic-team/vscode-extension#167]
# doc issues
After I installed many of your plugins to support my apps, three packages caused huge problems by making my entire node_modules directory lose its packages after running the `npm install` command. The problematic packages are:

```sh
@capacitor/google-maps
@capacitor/watch
@capacitor/background-runner
```

> ⚠️The issue is that they are both listed in the Ionic API documentation.⚠️

**Comment by dtarnawsky** [comment]
Hey @Bader-Idris ,

What do you mean by cause huge problems? Are you running `npm install` with `--force` in a project that isn't compatible?

I can help if you can provide a way to reproduce a problem.

**Comment by Bader-Idris** [comment]
because of using versions before 6 with documented APIs in ionic including the three I mentioned, then clicking on the migration to 6th capacitor version button, that caused conflicts, and these conflicts were removing all of node_modules dir, then I tried using `--force` and using `-legacy-peer-deps`, but after the completion of installation, it keeps removing all of the packages in node_modules

the problem is that the extension itself `ionic` does this, so I learned what is happening after deep searching of what extensions cause this, especially because there's no caution in the docs about the conflicts of the mentioned 3 packages above,

you can try installing the mock project of ionic then installing the three packages☝️, and then migrating to the 6th version, you'll encounter the conflicts I spoke of.

**Comment by dtarnawsky** [comment]
Ah interesting. The `cap migrate` was the missing piece. The extension will search through a project's plugins looking for plugins that will cause a peer dependency error, but it skips anything with @capacitor/ assuming that those would be taken care of by the `cap migrate` command. Unfortunately, all official plugins but those 3 plugins have been updated to Capacitor 6:
- Google Maps is waiting on an SPM module to be published
- Background runner is likely released next week
- Not sure on the release timeframe of the watch plugin

The cap migrate only has a set list of plugins it updates ([here](https://github.com/ionic-team/capacitor/blob/5b2ab7c68db4e77b8b2961e927acc5a708ee260f/cli/src/tasks/migrate.ts#L26)), so in v1.85.2 (just released now) I've matched to this list. So now the extension will report that these plugins are not compatible until **before** you migrate.

Thanks for reporting the bug!
